### ISSUE TITLE
Strange input issue with the Wii Classic Controller Pro shoulder buttons.

### ISSUE THREAD

**Issue by nl255** [KMFDManic/NESC-SNESC-Modifications#162]
### Step 1: Describe your setup:

  * hakchi2 CE version: 3.5_____
  * retroarch used (filename): 1.7.5 xtreme
  * hmod(s) used by the game you are having problens with (filenames): _____
  
### Step 2: Describe the problem:
  
When using a Wii Classic Controller Pro with Retroarch, at least on PSX games, pressing the left shoulder button will cause both l1 and l2 to be pressed.  The same with the right shoulder button and r1/r2.  However if I set l1 to zl, r1 to zr, l2 to left shoulder, and r2 to right shoulder in the global input settings (input user 1 binds) then everything works as expected.  Of course that means having to go into the core specific settings and set it so that l1=l2, r1=r2, l2=l1, r2=r1.

**Comment by KMFDManic** [comment]
You try turning left analog off in controls? That is usually how to contend with double controls for Cores like PPSSPP and PCSX ReArmed NEON.

**Comment by nl255** [comment]
I am not sure why that would make a difference, however adding the following lines to the global retroarch.cfg (and then changing the controls in the per-core overrides to compensate) did.  I still can't understand why the lines below stop the double input.

Double input free config

input_player1_l2_btn = "4"
input_player1_l_btn = "6"
input_player1_r2_btn = "5"
input_player1_r_btn = "7"


Double input happens

input_player1_l2_btn = "6"
input_player1_l_btn = "4"
input_player1_r2_btn = "7"
input_player1_r_btn = "5"

**Comment by KMFDManic** [comment]
I dont get double input. I am almost betting you have a knockoff controller.

**Comment by nl255** [comment]
It looks and feels like a Nintendo brand controller though I am using a 10 foot extension cord with it.  But if it was a controller issue why would the configuration change above make any difference?  I used the psx pad tester homebrew to verify that double input does not happen at all with that configuration change..

**Comment by KMFDManic** [comment]
I never had a single double input issue with any PS1 games with my controller that I got directly from Nintendo. But, generic ones, I've seen input issues happen with.  I actually have 3 pro classics and none of them give me double inputs.  These all came from Nintendo Store in New York. Your issue is one I've only seen others with generic controllers have. In fact, RetroArch even did an exact mapping of my controller that I saved, that is quite accurate to real Libretro mapping.  Was this a controller you got for roughly 10-20 USD?

**Comment by KMFDManic** [comment]
Did you by chance use another RetroArch previous to the one in my set? The controller mappings are a little different in Neo.

**Comment by nl255** [comment]
I might have but that was with a very old hakchi and I did format the user partition and then reset to factory defaults using hakchi ce between then and now.

The controller is a used one I got at a Disc Replay.

**Comment by KMFDManic** [comment]
Any chance you can post a picture of the serial, that good stuff from the controller?  I will do a look into the mapping configuration after work.  But, it is starting to seem that your controller is not using the proper Wii Pro Controller config.  I can verify this after work.  But, your picture will help determine whether controller is truly fake or not.  It is pointing to me as fake, at this point.  This is more common than you realize.  But, this can be a good thing as well as a bad thing.  Bad, if the controller is fake.  Good, because you and I can determine a special configuration that I can make into an HMOD for those that have fake controllers!

**Comment by nl255** [comment]
The only thing I could find on it is a sticker from Disc Replay and I am hesitant to remove it until the 30 day return/exchange period has passed.



As for custom configurations/hmods I made a guide on how to get everything fully working (except l3/r3) with the 8bit s/nesc retro receiver and a ps3 controller but when I posted it to /r/minisnesmods I found that I was the only one who could see it and even then only when logged in.  The same thing happened in /r/nesclassicmods as well.
